### ISSUE TITLE
Is it possible to do multispectral AND multitemporal at the same time?

### ISSUE THREAD

**Issue by MohammadElSakka** [danfenghong/IEEE_TPAMI_SpectralGPT#34]

**Comment by moonboy12138** [comment]
Thanks for your interest in our work! SpectralGPT is capable of handling both multispectral and multitemporal data. We have reserved the `t_patch_size` and `in_chans` parameters to support these modalities.

**Comment by MohammadElSakka** [comment]
Thank you for your work and your response! What should be the shape of the input when passing ```in_chans``` values != 1? So far I have tried passing (BATCH_SIZE, T, C, H, W), (BATCH_SIZE, T*C, H, W) but they do not seem to work. 

I suppose that ```t_patch_size``` should be equal to or divisible by C? and ```in_chans``` equal to the number of my observations ? 

Thank you in advance!

**Comment by moonboy12138** [comment]
We have provided the testing code in [this file](https://github.com/danfenghong/IEEE_TPAMI_SpectralGPT/blob/main/models_vit_tensor.py) at lines 370–374. You are right—the `t_patch_size` should be equal to or divisible by C, but `in_chans` does not necessarily need to equal the number of observations. All you need to do is debug the code to ensure it runs smoothly.

**Comment by MohammadElSakka** [comment]
If I understand correctly ```in_chans``` is used to set the number of inputs (2 in the case of Change Detection) is that right? So if I have 10 timestamps I could use ```in_chans = 10```?
In the other case, I could keep ```in_chans = 1``` and input 120 channels (12 spectral x 10 timestamps) with a ```t_patch_size = 12``` (number of channels per timestamp) ? If that's correct would setting  ```t_patch_size = 3``` or  ```t_patch_size = 4``` still allow the model to extract temporal features correctly?

**Comment by moonboy12138** [comment]
I understand what you need. You are right, and let me show you an example:

If `input = torch.rand(2, 12, 10, 120, 120)`, the model should be defined as follows:


```
def vit_base_patch8_120(**kwargs):
    model = VisionTransformer(
        img_size=120,
        in_chans=12,
        patch_size=8,
        embed_dim=768,
        depth=12,
        num_heads=12,
        mlp_ratio=4,
        num_frames=10,
        t_patch_size=10,
        norm_layer=partial(nn.LayerNorm, eps=1e-6),
        **kwargs,
    )
    return model
```
Here, `t_patch_size (set to 10)` can be reset to a value that is equal to or divisible by the number of timesteps, while in_chans should equal the number of your observations. Additionally, the code `x = torch.unsqueeze(x, dim=1)` is unnecessary in the forward procedure.

**Note:** Under these circumstances, the pretrained weight of SpectralGPT is no longer useful, as it does not take the temporal condition into account.

**Comment by MohammadElSakka** [comment]
Thank you for your answer! using these modifications, can I still use the provided weights of SpectralGPT+ to finetune on my own downstream task?

**Comment by moonboy12138** [comment]
Perhaps not; you can only use the provided weights of SpectralGPT+ for one timestamp image.

**Comment by MohammadElSakka** [comment]
Okay Thank you 👍
